### PR TITLE
Manual control result for cred hygiene control with reader permission

### PR DESCRIPTION
**Pull request by abhaydaga**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 [hunk 1]
```diff
@@ -1527,82 +1527,89 @@ class SubscriptionCore: AzSVTBase
 
 	hidden [ControlResult] CheckCredentialHygiene([ControlResult] $controlResult)
     {
-        $AzSKRG = [ConfigurationManager]::GetAzSKConfigData().AzSKRGName
+		$AzSKRG = [ConfigurationManager]::GetAzSKConfigData().AzSKRGName
```
Comment (VishalHaibatpure) [inline src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: code/logic comments missing all over the function

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 [hunk 2]
```diff
@@ -1527,82 +1527,89 @@ class SubscriptionCore: AzSVTBase
...
         $containerName = [Constants]::RotationMetadataContainerName
         $StorageAccount = Get-AzStorageAccount -ResourceGroupName $AzSKRG | Where-Object {$_.StorageAccountName -like 'azsk*'} -ErrorAction SilentlyContinue
```
Comment (VishalHaibatpure) [inline src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: I think we already have helper method to AzSK storage account Plz leverage same.

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 [hunk 3]
```diff
@@ -1527,82 +1527,89 @@ class SubscriptionCore: AzSVTBase
...
+					}
+				}
 
-			if($expiredCount -gt 0){
-				$controlResult.VerificationResult = [VerificationResult]::Failed;
-				$controlResult.AddMessage("`nPlease update them soon using the cmd Update-AzSKTrackedCredential with the 'ResetLastUpdate' switch with other required parameters (Subscription Id, credential name, etc.).`n")
-			}
-			elseif($aboutToExpireCount -gt 0){
-				$controlResult.VerificationResult = [VerificationResult]::Verify
+				$controlResult.AddMessage("`nCredentials that have expired or are very close to expiry: $expiredCount `n", $expiredCredentials)
```
Comment (VishalHaibatpure) [inline src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: Discuss these message with Manish.
Credentials that are not near expiry can become "Credentials that are healthy"

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 [hunk 4]
```diff
@@ -1527,82 +1527,89 @@ class SubscriptionCore: AzSVTBase
 
 	hidden [ControlResult] CheckCredentialHygiene([ControlResult] $controlResult)
     {
-        $AzSKRG = [ConfigurationManager]::GetAzSKConfigData().AzSKRGName
+		$AzSKRG = [ConfigurationManager]::GetAzSKConfigData().AzSKRGName
         $containerName = [Constants]::RotationMetadataContainerName
         $StorageAccount = Get-AzStorageAccount -ResourceGroupName $AzSKRG | Where-Object {$_.StorageAccountName -like 'azsk*'} -ErrorAction SilentlyContinue
         $keys = Get-AzStorageAccountKey -ResourceGroupName $AzSKRG -Name $StorageAccount.StorageAccountName -ErrorAction SilentlyContinue
```
Comment (VishalHaibatpure) [inline src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: This line will break for subs where AzSK storage account is not present

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 [hunk 5]
```diff
@@ -1527,82 +1527,89 @@ class SubscriptionCore: AzSVTBase
...
-		if($container){
-			$credBlobs = $container | Get-AzStorageBlob
-
-			$expiredCount = 0;
-			$aboutToExpireCount = 0;
-			$healthyCount = 0;
-			[PSObject] $expiredCredentials = @();
-			[PSObject] $aboutToExpireCredentials = @();
-			[PSObject] $healthyCredentials = @();
+		if($keys)
```
Comment (VishalHaibatpure) [inline src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: We should try to leverage HasAttestationReadPermission flag with the help of method
$this.ControlStateExt.HasControlStateReadAccessPermissions()

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 (hunk 3 above)
Comment (abhaydaga) [reply src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: Messages have been already discussed.

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 (hunk 5 above)
Comment (abhaydaga) [reply src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: yes, it wasn't working. perhaps it is set once the code flow reaches the attestation workflow.

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 [hunk 6]
```diff
@@ -1527,82 +1527,89 @@ class SubscriptionCore: AzSVTBase
...
-				$expiryTime = $lastRotatedTime.AddDays($credentialInfo.rotationInt);
-				if($expiryTime -le $currentTime.AddDays($this.ControlSettings.SubscriptionCore.credHighTH)){
-					$expiredCount += 1;
-					$expiredCredentials += $credentialInfo;
-				}
-				elseif(($expiryTime -gt $currentTime.AddDays($this.ControlSettings.SubscriptionCore.credHighTH)) -and ($expiryTime -le $currentTime.AddDays($this.ControlSettings.SubscriptionCore.credModerateTH))){
-					$aboutToExpireCount +=1;
-					$aboutToExpireCredentials += $credentialInfo;
+				if(![string]::isnullorwhitespace($this.SubscriptionContext.SubscriptionId)){
+					if(-not (Test-Path $tempSubPath))
```
Comment (VishalHaibatpure) [inline src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: Check if we have helper method for create AzSK Temp folder. If no, we should have it.

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 [hunk 7]
```diff
@@ -1527,82 +1527,89 @@ class SubscriptionCore: AzSVTBase
...
 				}
-			}
 
-			$controlResult.AddMessage("`nCredentials that have expired or are very close to expiry: $expiredCount `n", $expiredCredentials)
-			$controlResult.AddMessage("`nCredentials that are approaching expiry: $aboutToExpireCount `n", $aboutToExpireCredentials)
-			$controlResult.AddMessage("`nCredentials that are not near expiry: $healthyCount `n", $healthyCredentials)
+				$credBlobs | ForEach-Object{
+					$file = Join-Path $AzSKTemp -ChildPath $($this.SubscriptionContext.SubscriptionId) | Join-Path -ChildPath $($_.Name)
+					
+					$blobContent = Get-AzStorageBlobContent -Blob $_.Name -Container $container.Name -Context $context -Destination $file -Force -ErrorAction Ignore
```
Comment (VishalHaibatpure) [inline src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: Try using StorageHelper for getting blob contents

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 (hunk 6 above)
Comment (abhaydaga) [reply src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: We dont have a helper function.

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 (hunk 1 above)
Comment (abhaydaga) [reply src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: done

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 (hunk 7 above)
Comment (abhaydaga) [reply src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: Storage helper cmd has code which is not compatible with PS Core.

 static [object] GetStorageBlobContent([string] $folderName, [string] $fileName, [string] $blobName, [string] $containerName, [object] $stgCtx)
	{
             $fileName = $folderName.TrimEnd("\")+ "\" + $fileName
             return [StorageHelper]::GetStorageBlobContent($fileName, $blobName, $containerName, $stgCtx)
    }

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 (hunk 4 above)
Comment (abhaydaga) [reply src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: Yes. This is tracked by me. Will push these changes in the next build.

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 (hunk 2 above)
Comment (abhaydaga) [reply src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: Using the helper command will end up spinning a storage account & container which we dont want atleast in the control logic.  We are checking of the container exists or not. Anyways, handling of AzSKRG resources will be handled in SSS & USS.

**Comment by abhaydaga** [comment]
code moved to another branch

**Pull request closed without merging**